File: 7-kyu/balanced-number-(special-numbers-series-#1-)-/python/solution.py
def balanced_num(number):
    number_str = str(number)
    if len(number_str) <= 2:
        return 'Balanced'
    middle = len(number_str) // 2
    if len(number_str) % 2 == 0: middle -= 1
    lhs = list(map(int, number_str[:middle]))
    rhs = list(map(int, number_str[len(number_str) - middle:]))
    return 'Balanced' if sum(lhs) == sum(rhs) else 'Not Balanced'

File: 7-kyu/balanced-number-(special-numbers-series-#1-)-/c++/solution.cpp
#include <numeric>
#include <string>
using namespace std;

int getSum(string number_str)
{
  return accumulate(number_str.begin(), number_str.end(), 0,
    [](int sum, char c) { return sum + (c - '0'); });
}

string balancedNum (unsigned long long int number )
{
  auto number_str = to_string(number);
  if (number_str.length() <= 2) return "Balanced";
  string left = "", right = "";
  auto middle = number_str.length() / 2;
  if (number_str.length() % 2 == 0) middle--;
  left = number_str.substr(0, middle);
  right = number_str.substr(number_str.length() - middle);
  return getSum(left) == getSum(right) ? "Balanced" : "Not Balanced";
}